WordPress/src/main/java/org/wordpress/android/ui/posts/RemotePreviewLogicHelper.kt

```diff
@@ -93,14 +98,11 @@ class RemotePreviewLogicHelper @Inject constructor(
         return PreviewLogicOperationResult.GENERATING_PREVIEW
     }
 
-    private fun shouldUpload(post: PostModel): Boolean {
-        val status = PostStatus.fromPost(post)
-        return post.isLocalDraft ||
-                (status == PostStatus.DRAFT && post.isLocallyChanged)
+    private fun shouldUpload(post: PostModel, action: PostUploadAction): Boolean {
+        return (post.isLocallyChanged || post.isLocalDraft) && (action == UPLOAD_AS_DRAFT || action == UPLOAD)
```

Comment:
The UPLOAD action is listed here for an edge case scenario when the user confirms the changes but the post doesn't get uploaded and they click on Remote Preview button soon after.